vimiv/gui/metadatawidget.py

```diff
@@ -54,20 +55,42 @@ def __init__(self, parent):
             self._mainwindow_bottom = 0
             self._mainwindow_width = 0
             self._path = ""
-            self._loaded = False
+            self._current_set = ""
 
             api.signals.new_image_opened.connect(self._on_image_opened)
+            api.settings.metadata.current_keyset.changed.connect(self._update_text)
 
             self.hide()
 
         @api.keybindings.register("i", "metadata", mode=api.modes.IMAGE)
         @api.commands.register(mode=api.modes.IMAGE)
-        def metadata(self):
-            """Toggle display of exif metadata of current image."""
+        def metadata(self, count: Optional[int] = None):
+            """Toggle display of exif metadata of current image and handle switching of metadata keysets.
+
+            **count:** Select the key set to display.
+            """
+
             if self.isVisible():
-                _logger.debug("Hiding widget")
-                self.hide()
+                if count is not None:
+                    if self._count_is_valid(count):
+                        _logger.debug("Switch keyset in open widget")
+                        api.settings.metadata.current_keyset.value = api.settings.metadata.keysets[
+                            count - 1
+                        ]
+                        self._update_text()
```

Comment:
This call to `self._update_text()` should not be needed thanks to the signal connection you did in the constructor.